test/docfx.Test/e2e/E2ETest.cs

```diff
@@ -44,28 +44,34 @@ private static async Task RunCore(string docsetPath, E2ESpec spec)
                 await Program.Run(command.Split(" ").Concat(new[] { docsetPath }).ToArray());
             }
 
-            // Verify restored files
-            foreach (var (file, content) in spec.Restores)
-            {
-                var restoredFile = Directory.EnumerateFiles(AppData.AppDataDir, file, SearchOption.TopDirectoryOnly).FirstOrDefault();
-                Assert.NotNull(restoredFile);
-                Assert.True(File.Exists(restoredFile));
-                VerifyFile(restoredFile, content);
-            }
-
             // Verify output
             var docsetOutputPath = Path.Combine(docsetPath, "_site");
             Assert.True(Directory.Exists(docsetOutputPath));
 
             var outputs = Directory.GetFiles(docsetOutputPath, "*", SearchOption.AllDirectories);
             var outputFileNames = outputs.Select(file => file.Substring(docsetOutputPath.Length + 1).Replace('\\', '/')).ToList();
 
+            // Show build.log content if actual output has errors or warnings.
+            if (!spec.Outputs.Keys.Contains("build.log") && outputFileNames.Contains("build.log"))
+            {
+                Assert.True(false, File.ReadAllText(Path.Combine(docsetOutputPath, "build.log")));
+            }
+
             Assert.Equal(spec.Outputs.Keys.OrderBy(_ => _), outputFileNames.OrderBy(_ => _));
 
             foreach (var (filename, content) in spec.Outputs)
             {
                 VerifyFile(Path.GetFullPath(Path.Combine(docsetOutputPath, filename)), content);
             }
+
+            // Verify restored files
```

Comment:
why move to last？ If the restoring failed, the output file may also be wrong, then you can't easily found the root cause